clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -1952,19 +1942,22 @@ private void assertStaleMemberLeavesGroupAndClearsAssignment(MembershipManagerIm
         // Should reset epoch to leave the group and release the assignment (right away because
         // there is no onPartitionsLost callback defined)
         verify(subscriptionState).assignFromSubscribed(Collections.emptySet());
-        assertTrue(membershipManager.currentAssignment().isEmpty());
+        assertTrue(membershipManager.currentAssignment().isNone());
         assertTrue(membershipManager.topicsAwaitingReconciliation().isEmpty());
         assertEquals(LEAVE_GROUP_MEMBER_EPOCH, membershipManager.memberEpoch());
     }
 
     @Test
     public void testMemberJoiningTransitionsToStableWhenReceivingEmptyAssignment() {
```

Comment:
I cannot find any test for one of the 2 issues we're after with this PR: ensure that we are triggering the callbacks when joining and getting empty assignment. Could we add it? I expect it should be very similar to this one, but just providing a `CounterConsumerRebalanceListener` and asserting that it called the `onPartitionsAssigned`

Reply:
I wonder if we could also add a test for the case described in KAFKA-16185. Is it possible?